File: ruby/ruby/thread.rb
require 'thread'

def countdown(name)
    i = 9
    while i > -1
        puts "#{name}: #{i}" 
        sleep 1
        i = i - 1
    end
end

def countup(name)
    for i in (0..9)
        puts "#{name}: #{i}" 
        sleep 1
    end
end


t1 = Thread.new{countdown('THREAD #01')}
t2 = Thread.new{countup('THREAD #02')}
t1.join
t2.join

File: python/Study 02/python02/thread.py
import time
import threading


def CountDown(*args):
    i = 10
    for i in range (10, 0, -1):
        print("THREAD #01: %d" %(i))
        time.sleep(1)
    print("\n\nTHREAD #01 completo!!!")

def CountUp(*args):
    i = 0
    for i in range (0, 10, 1):
        print("\n\nTHREAD #02: %d" %(i))
        time.sleep(1)
    print("THREAD #02 completo!!!")
   
t1 = threading.Thread(target=CountDown())
t2 = threading.Thread(target=CountUp())   
# CountDown()
# CountUp() 
t1.start()
t2.start()